src/megaapi_impl.cpp

```diff
@@ -23107,6 +23148,128 @@ void MegaApiImpl::sendPendingRequests()
+            if (!tlvRecords || !tlvRecords->find("h")) { e = API_ENOENT; break; }
+
+            const string& handleStr = tlvRecords->get("h");
+            handle h = MegaApi::base64ToHandle(handleStr.c_str());
+            if (h == UNDEF) { e = API_ENOENT; break; }
+
+            // get Node of remote "My Backups" folder
+            MegaNode* myBackupsNode = getNodeByHandle(h);
+            if (!myBackupsNode) { e = API_ENOENT; break; }
+
+            // get `DEVICE_NAME`, from user attributes
+            if (!u->isattrvalid(ATTR_DEVICE_NAMES)) { e = API_EINCOMPLETE; break; }
+            const string* deviceNameContainerStr = u->getattr(ATTR_DEVICE_NAMES);
+            if (!deviceNameContainerStr) { e = API_EINCOMPLETE; break; }
+
+            tlvRecords.reset(TLVstore::containerToTLVrecords(deviceNameContainerStr, &client->key));
+            if (!tlvRecords || !tlvRecords->find("n")) { e = API_EINCOMPLETE; break; } // is 'n' the right key ??
+
+            const string& devName64Str = tlvRecords->get("n");
+            string deviceName = Base64::btoa(devName64Str);
+            if (deviceName.empty()) { e = API_EINCOMPLETE; break; }
+
+            // get 'device-id'
+            const string& deviceId = client->getDeviceid();
+            if (deviceId.empty()) { e = API_EINCOMPLETE; break; }
+
+            // loop remote children of "My Backups" folder, to find `DEVICE_NAME`-with-'device-id' node
+            const MegaNodeList* nodeList = myBackupsNode->getChildren();
+            MegaNode* deviceNameNode = nullptr;
+            for (int i = 0; i < nodeList->size(); ++i)
+            {
+                MegaNode* child = nodeList->get(i);
+
+                // get folder name
+                const char* name = child->getName();
+                bool gotName = deviceName == name;
+
+                // get 'device-id' tag
+                const char* id = child->getCustomAttr("device-id");
+                bool gotId = deviceId == id;
+
+                // determine whether we found our node or not
+                if (!gotName && !gotId) { continue; } // ignore this child folder
+
+                if (gotName && gotId) // got it, use this one to add another backup to it
+                {
+                    deviceNameNode = child;
+                    break;
+                }
+
+                // there's an error if only one of the folder's {name, id} is correct and the other is wrong.
+                e = API_EINCOMPLETE;
+                break;
+            }
+
+            if (e != API_OK) { break; }
+
+            if (!deviceNameNode)
+            {
+                // create "My Backups"/`DEVICE_NAME`/
+                // HOW ? Like this?
+                auto nextSyncTag = client->nextSyncTag();
+                auto sync = make_unique<MegaSyncPrivate>(nullptr, deviceName.c_str(), h, nextSyncTag);
+                sync->setListener(request->getSyncListener());
+
+                // NOW WHAT ? Something like this?
```

Comment:
Yes, I think that's the idea :+1:

Reply:
So, I should "not create a sync" (as noted in the previous comment), but `client->addsync()` does take a `SyncConfig` as first, and a `MegaSyncPrivate` as 6th argument. I'm failing to match the names with the arguments and the purpose of such calls...

Reply:
In my previous comment I was confused by looking at the comment, I was replying to the comment: 
```
// create "My Backups"/`DEVICE_NAME`/
// HOW ? Like this?
```
Creating a `SyncConfig` and a `Sync` is not the way to create such folder. Hence my comment.
But the final goal of this new interface is to create a sync, so obviously at some point you need to create the aforementioned objects.